Planner/app/src/main/java/rdatu/android/cyscorpions/com/projectplanner/view/DatePickerDialog.java

```diff
@@ -0,0 +1,93 @@
+package rdatu.android.cyscorpions.com.projectplanner.view;
+
+import android.app.Activity;
+import android.app.Dialog;
+import android.os.Bundle;
+import android.support.v4.app.DialogFragment;
+import android.util.Log;
+import android.widget.DatePicker;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * Created by rayeldatu on 7/28/15.
+ */
+public class DatePickerDialog extends DialogFragment implements android.app.DatePickerDialog.OnDateSetListener {
+
+    public static final String EXTRA_DATE = "date";
+    public static final String EXTRA_FUNCTION = "function";
+    private Callbacks mCallbacks;
+    private String mStringDate, mDatePickerFunction;
+
+    public DatePickerDialog() {
+
+    }
+
+    public static DatePickerDialog newInstance(String date, String function) {
+        DatePickerDialog fragment = new DatePickerDialog();
+        Bundle args = new Bundle();
+        args.putSerializable(EXTRA_FUNCTION, function);
+        args.putSerializable(EXTRA_DATE, date);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mStringDate = getArguments().getString(EXTRA_DATE);
+        mDatePickerFunction = getArguments().getString(EXTRA_FUNCTION);
+    }
+
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+        mCallbacks = (Callbacks) activity;
+    }
+
+    @Override
+    public void onDateSet(DatePicker view, int year, int monthOfYear, int dayOfMonth) {
+        SimpleDateFormat df = new SimpleDateFormat("MMM-dd-yyyy");
+        Calendar c = Calendar.getInstance();
+        c.set(Calendar.YEAR, year);
+        c.set(Calendar.MONTH, monthOfYear);
+        c.set(Calendar.DAY_OF_MONTH, dayOfMonth);
+        switch (mDatePickerFunction) {
```

Comment:
Minor: *Optional* *For Backwards Compatibility*
In Java 1.6 Environment, we cannot use String for switch-case statements. This will result to compile error